packages/vm/src/calls.rs

```diff
@@ -86,6 +101,21 @@ pub fn call_handle_raw<S: Storage + 'static, A: Api + 'static, Q: Querier + 'sta
     call_raw(instance, "handle", &[env, msg], MAX_LENGTH_HANDLE)
 }
 
+/// Calls Wasm export "migrate" and returns raw data from the contract.
+/// The result is length limited to prevent abuse but otherwise unchecked.
+pub fn call_migrate_raw<S: Storage + 'static, A: Api + 'static, Q: Querier + 'static>(
+    instance: &mut Instance<S, A, Q>,
+    env: &[u8],
+    msg: &[u8],
+) -> VmResult<Vec<u8>> {
+    // first check if migrate is not implemented and return a standard error message
+    if instance.func::<(u32, u32), u32>("migrate").is_err() {
```

Comment:
This error case is handles automatically in `call_raw`:

```rust
        2 => {
            let func: Func<(u32, u32), u32> = instance.func(name)?;
            func.call(arg_region_ptrs[0], arg_region_ptrs[1])?
        }
```

with

```rust
    pub(crate) fn func<Args, Rets>(&self, name: &str) -> VmResult<Func<Args, Rets, Wasm>>
    where
        Args: WasmTypeList,
        Rets: WasmTypeList,
    {
        let function = self.inner.exports.get(name)?;
        Ok(function)
    }
```

and
 
```rust
impl From<wasmer_runtime_core::error::ResolveError> for VmError {
    fn from(original: wasmer_runtime_core::error::ResolveError) -> Self {
        make_resolve_err(format!("Wasmer resolve error: {:?}", original))
    }
}
```

Do we really need to handle it separately here?

Reply:
Ah, I didn't realize this was gracefully handled. Nice catch. I will simplify the code.

Reply:
Fixed in e64d217